### PR TITLE
Calendar events changes

### PR DESCRIPTION
**Pull request by bobbykolev**
https://github.com/alkem-io/client-web/issues/3912 - Create/Update 
https://github.com/alkem-io/client-web/issues/6874 - Display

Dialog (create/edit) to be updated as per design
 - [x] UI to match the design
   - [x] name & type on first line;
   - [x] end date and time instead of duration;
   - [x] all day switch; 
 - [x] All-day event (wholeDay field);
 - [x] 15 min time steps instead of 30;
 - [x] Add location field;
 - [x] Multi-day display support
   - [x] Rounded badge 
   - [x] event details header;
   - [x] calendar; 
- [x] Now the event description is no longer required (with no default value); 
 - Add references (after temp storage, to be extracted);
 - Add URL field - to be extracted;

Technical
- [x] multi-day validation;

Create/Edit
Screenshot 2024-10-08 181353
Screenshot 2024-10-10 133524
Screenshot 2024-10-10 155944

## Summary by CodeRabbit

- **New Features**
	- Enhanced localization with updated translations for Bulgarian, German, English, Spanish, French, Dutch, Portuguese, and Ukrainian.
	- Added new time slots in the DatePicker for improved time selection.
	- Introduced new properties in calendar event forms for better event management, including `endDate` and `location`.

- **Bug Fixes**
	- Improved clarity and consistency in notifications and user messages across multiple languages.

- **Enhancements**
	- Updated components for better functionality and responsiveness, including event highlighting and duration handling in calendar views.
	- Enhanced date handling and formatting functionalities for improved user experience.
	- Refined cookie consent messages for better user understanding.

### CODE REVIEW

File: src/core/utils/time/utils.ts [hunk 1]
```diff
@@ -99,3 +182,18 @@ export const startOfDay = (date: Date = new Date()) => {
...
+
+  const startDateOnly = new Date(startDate);
+  const endDateOnly = new Date(endDate);
+
+  return (
+    startDateOnly.getFullYear() === endDateOnly.getFullYear() &&
+    startDateOnly.getMonth() === endDateOnly.getMonth() &&
+    startDateOnly.getDate() === endDateOnly.getDate()
+  );
+};
```
[the remark this answers is not in the document]
Comment (valeksiev) [reply src/core/utils/time/utils.ts:199]: Alternatively, formatting the two dates with the same locale and comparing them will be more readable.

**Review by valeksiev — APPROVED** [review]
Overall, I agree with the Bunny's comments so at least review them thoroughly.

Approving as I don't feel any of them is crucial and don't want to be blocking.

**Comment by valeksiev** [comment]
image
this is happening - choose a start date after the end date, then choose `All-day`. Also, 

image

When I close the calendar, the date is reset to 14.10

image
I was not notified that I couldn't create the event because the start and end times are the same.

File: src/domain/timeline/calendar/components/FullCalendar.tsx [hunk 2]
```diff
@@ -179,34 +190,53 @@ interface EventsTooltipProps {
...
-    <Box className="tooltip-anchor" />
-  </Tooltip>
-);
+
+    if (event.durationDays && event.durationMinutes) {
+      return name;
+    }
+
+    return `${dayjs(event.startDate).format('HH:mm')} - ${name}`;
+  };
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/domain/timeline/calendar/components/FullCalendar.tsx:212]: There should be no events without startDate. However, the type is defined as optional on the server. I'm going to resolve this for now.

**Comment by bobbykolev** [comment]
> export const isSameDay = (startDate: Date | undefined, endDate: number | Date | undefined) => {
>   if (!startDate || !endDate) {
>     return true;
>   }
> 
>   const startDateOnly = new Date(startDate);
>   const endDateOnly = new Date(endDate);
> 
>   return (
>     startDateOnly.getFullYear() === endDateOnly.getFullYear() &&
>     startDateOnly.getMonth() === endDateOnly.getMonth() &&
>     startDateOnly.getDate() === endDateOnly.getDate()
>   );
> };

Yes, this is hard to handle (can't think of something ATM) so let's extract it in bug so we're not delaying this and other priorities.
@Comoque1 FYI

File: src/domain/timeline/calendar/views/CalendarEventsList.tsx [hunk 3]
```diff
@@ -26,6 +26,8 @@ interface CalendarEventsListProps {
     id: string;
     nameID: string;
     startDate?: Date;
+    durationDays?: number | undefined;
+    durationMinutes: number;
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/domain/timeline/calendar/views/CalendarEventsList.tsx:30]: Yes, no need to be processed there as it is needed and processed in its children.

File: src/core/utils/time/utils.ts [hunk 4]
```diff
@@ -12,33 +12,97 @@ const LocaleId = 'en-GB';
...
+      startDate: startDateFormatted,
+      endDate: endDate.toLocaleDateString(LocaleId, { day: '2-digit', month: '2-digit' }),
+    };
+  }
+
+  return {
+    startDate: startDateFormatted,
+    endDate: null,
+  };
+};
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/core/utils/time/utils.ts:65]: If durationMinutes are not specified then this is not a valid event.

File: src/domain/timeline/calendar/views/CalendarEventForm.tsx [hunk 5]
```diff
@@ -88,66 +106,150 @@ const CalendarEventForm = ({
...
+      }),
+    endDate: yup
+      .date()
+      .nullable()
+      .test('valid-endDate', 'Invalid end date', function (value) {
+        return validateDuration({ ...this.parent, endDate: value });
+      }),
+    wholeDay: yup.boolean().test('valid-wholeDay', 'Invalid whole day', function (value) {
+      return validateDuration({ ...this.parent, wholeDay: value });
+    }),
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/domain/timeline/calendar/views/CalendarEventForm.tsx:155]: It was like that, but for some reason, when switching back and forth the all-day event with incorrect endDate, the form stayed valid.

**Review by valeksiev — DISMISSED** [review]

**Review by valeksiev — APPROVED** [review]

**Pull request merged**